File: src/main/java/com/gpch/hotel/model/Product.java
package com.gpch.hotel.model;


import com.fasterxml.jackson.annotation.JsonBackReference;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "product")
public class Product {
    @Id
    @Column(name = "product_id")
    private long id;
    @Column(name = "productname")
    private String productName;
    @Column(name = "price")
    private int price;
    @Column(name = "amount")
    private int amount;
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name="store_id", nullable=false)
    private Store stores;


    public long getIdStore() {
        return stores.getId();
    }

}

File: src/main/kotlin/com/gpch/hotel/model/Product.kt
package com.gpch.hotel.model

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
@Table(name = "product")
class Product {

    @Id
    @Column(name = "product_id")
    var id: Long = 0

    @Column(name = "productname")
    var productName: String? = null

    @Column(name = "price")
    var price: Int = 0

    @Column(name = "amount")
    var amount: Int = 0

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "store_id", nullable = false)
    var stores: Store? = null

    fun getIdStore(): Long = stores?.id ?: 0L
}
